src/NServiceBus.Extensions.Hosting/WebHostCompatibleMessageSession.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace NServiceBus.Extensions.Hosting
+{
+    using System;
+    using System.Threading.Tasks;
+
+    class WebHostCompatibleMessageSession : IMessageSession
+    {
+        readonly NServiceBusHostedService hostedService;
+
+        public WebHostCompatibleMessageSession(NServiceBusHostedService hostedService)
+        {
+            this.hostedService = hostedService;
+        }
+
+        public async Task Send(object message, SendOptions options)
+        {
+            var session = await hostedService.Endpoint.ConfigureAwait(false);
```

Comment:
So if the hosted service's `StartAsync()` has not finished and the `Endpoint` property was not assigned yet, web host message session operation will await and get the result (or throw an exception). Nice.